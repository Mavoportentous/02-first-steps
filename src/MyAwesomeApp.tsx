import { useState } from "react";
import type { CSSProperties } from "react";
const firstName = 'Miguel Angel';
const LastName = 'Valenzuela Ortiz';
const favoriteGames = ['League of Legends', 'Control', 'Stronghold Crusader', 'Streonghold', 'Age of Mithology', 'Ades'];
let isActive = true;//los valores boleanos no generan código interno en react
const addres = {
    zipCode: '54870',
    country: 'México'
}


export const MyFunctionClick = () => {

    if (isActive === true) { isActive = false }
    else {
        isActive = true
    }
    console.log("desdeMyFunctionClick " + isActive);
};

export const MyAwesomeApp = () => {
    //Echo con ayuda de Gemini 2.5 Flash
    // 1. Usa useState para crear una variable de estado y su función de actualización
    const [isActive, setIsActive] = useState(true);

    // 2. Define la función de alternancia dentro del componente
    const toggleActive = () => {
        // Usa el setter para actualizar el estado al valor opuesto
        // React detecta este cambio y re-renderiza el componente
        setIsActive(!isActive);
        console.log("El nuevo estado es: " + !isActive);
    };

    // 3. Define los estilos, ya que dependen del estado 'isActive'
    const styles: CSSProperties = {
        // Nota: El formato hexadecimal de 4 dígitos es #RGBA. 
        // Si quieres un blanco, usa '#FFFF' o 'white'. Asumo que buscabas 'rgba(250, 250, 250, 1)'
        backgroundColor: isActive ? '#FAFAFA' : 'red',
        borderRadius: 20,
        padding: 10,
        borderColor: 'blue'
    };


    return (
        <div data-testid="div-app">
            <h1 data-testid='first-name-title'>{firstName}</h1>
            <h3>{LastName}</h3>
            <p>{favoriteGames.join(', ')}</p>

            {/* Renderizado condicional basado en el estado */}
            <h1 data-testid='h1-active-desactive'>{isActive ? 'Activo ✅' : 'No activo ❌'}</h1>

            <p>{addres.country + ' ' + addres.zipCode}</p>

            <p style={styles}>{JSON.stringify(addres)}</p>

            {/* 4. Asigna la función de alternancia al evento onClick. 
                 ¡Importante! Se pasa la referencia sin paréntesis. */}
            <button onClick={toggleActive}>
                Activar | Desactivar
            </button>
        </div>
    );
};
// export function MyAwesomeApp() {
//     return (
//         <>
//             <h1>Miguel</h1>
//             <h3>Valenzuela</h3>
//         </>
//     );

// }