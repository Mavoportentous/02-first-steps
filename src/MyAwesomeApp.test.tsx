import { describe, expect, test } from 'vitest';
import { fireEvent, render, screen } from '@testing-library/react';

import { MyAwesomeApp } from './MyAwesomeApp';

describe('MyAwesomeApp', () => {
  test('should render firstName and lastName', () => {
    const { container } = render(<MyAwesomeApp />);
    // screen.debug();
    // console.log(container.innerHTML);

    const h1 = container.querySelector('h1');
    const h3 = container.querySelector('h3');

    expect(h1?.innerHTML).toContain('Miguel');
    expect(h3?.innerHTML).toContain('Valenzuela');
  });

  test('should render firstName and lastName - screen', () => {
    render(<MyAwesomeApp />);
    screen.debug();
    // console.log(container.innerHTML);

    // const h1 = screen.getByRole('heading', {
    //   level: 1,
    // });
    const h1 = screen.getByTestId('first-name-title');
    expect(h1.innerHTML).toContain('Miguel');
  });

  // test('should match snapshot', () => {
  //   const { container } = render(<MyAwesomeApp />);
  //   expect(container).toMatchSnapshot();
  // });

  // test('should match snapshot', () => {
  //   render(<MyAwesomeApp />);
  //   expect(screen.getByTestId('div-app')).toMatchSnapshot();
  // });

  test('should test my  MyFunctionClick', () => {
    render(<MyAwesomeApp />);

    const [btn] = screen.getAllByRole('button');

    fireEvent.click(btn);

    expect(screen.getByText('No activo ❌')).toBeDefined();

  })
});

describe('MyFunctionClick', () => {
  test('shold show the value of isActive', () => {
    const isActive = false;

    expect(isActive != false);

  });


});