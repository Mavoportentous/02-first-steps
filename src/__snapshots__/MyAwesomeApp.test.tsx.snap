// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h1
      data-testid="first-name-title"
    >
      Miguel Angel
    </h1>
    <h3>
      Valenzuela Ortiz
    </h3>
    <p>
      League of Legends, Control, Stronghold Crusader, Streonghold, Age of Mithology, Ades
    </p>
    <h1>
      Activo ✅
    </h1>
    <p>
      México 54870
    </p>
    <p
      style="background-color: rgb(250, 250, 250); border-radius: 20px; padding: 10px; border-color: blue;"
    >
      {"zipCode":"54870","country":"México"}
    </p>
    <button>
      Activar | Desactivar
    </button>
  </div>
</div>
`;

exports[`MyAwesomeApp > should match snapshot 2`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-title"
  >
    Miguel Angel
  </h1>
  <h3>
    Valenzuela Ortiz
  </h3>
  <p>
    League of Legends, Control, Stronghold Crusader, Streonghold, Age of Mithology, Ades
  </p>
  <h1>
    Activo ✅
  </h1>
  <p>
    México 54870
  </p>
  <p
    style="background-color: rgb(250, 250, 250); border-radius: 20px; padding: 10px; border-color: blue;"
  >
    {"zipCode":"54870","country":"México"}
  </p>
  <button>
    Activar | Desactivar
  </button>
</div>
`;
